pkg/providers/instance/instance.go

```diff
@@ -670,6 +705,54 @@ func (p *Provider) getAKSIdentifyingExtension() *armcompute.VirtualMachineExtens
 	return vmExtension
 }
 
+func (p *Provider) getCSExtension(cse string, isWindows bool) *armcompute.VirtualMachineExtension {
+	const (
+		vmExtensionType     = "Microsoft.Compute/virtualMachines/extensions"
+		cseNameWindows      = "windows-cse-agent-karpenter"
```

Comment:
why is there a unique name for karpenter CSE? Would this break any assumptions made https://www.hyrumslaw.com/

Reply:
For consistency.